### PR TITLE
Rule design optimization

### PR DESCRIPTION
**Pull request by iproger**
### Description
Refactor rules: remove excess structures and optimize.
Replace sizeof() to more fast count().

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18436) <br/>All committers have signed the CLA.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-3110](https://jira.corp.magento.com/browse/ENGCOM-3110) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@iproger thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: app/code/Magento/Catalog/Block/Adminhtml/Category/AbstractCategory.php [hunk 1]
```diff
@@ -102,6 +102,7 @@ public function getCategoryPath()
     public function hasStoreRootCategory()
     {
         $root = $this->getRoot();
+        return $root && $root->getId();
         if ($root && $root->getId()) {
```
Comment (hostep) [inline app/code/Magento/Catalog/Block/Adminhtml/Category/AbstractCategory.php]: I think these lines can be removed, right?

Comment (iproger) [reply app/code/Magento/Catalog/Block/Adminhtml/Category/AbstractCategory.php]: @hostep Yes, I added fix commit.

**Comment by ihor-sviziev** [comment]
Hi @iproger,
Right now static tests are failing for files you've changed, could you adjust code to pass static tests?

**Comment by iproger** [comment]
@ihor-sviziev I'll try.

**Comment by sidolov** [comment]
Hi @iproger , will you continue work on this PR?

**Comment by sidolov** [comment]
Hi @iproger , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**